apps/demos/Demos/Chat/Overview/React/App.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useState } from 'react';
+
+import { Chat, ChatTypes } from 'devextreme-react/chat';
+import { MessageEnteredEvent } from 'devextreme/ui/chat';
+
+import service from './data.ts';
+
+export default function App() {
+    const [ userChatTypingUsers, setUserChatTypingUsers ] = useState([]);
+    const [ supportChatTypingUsers, setSupportChatTypingUsers ] = useState([]);
+    const [ messages, setMessages ] = useState<ChatTypes.Message[]>(service.messages);
+
+    function onMessageEntered(event: MessageEnteredEvent) {
+        setMessages(prevMessages => [...prevMessages, event.message]);
+    }
+
+    function userChatTypingStart() {
```

Comment:
I would prefer to have a single handler for both typingStart and typingEnd events for the Chat. It could look like this
```
    function onTypingStart({ user }) {
        if (user.id === currentUser.id) {
            setUserChatTypingUsers([currentUser]);
        } else {
            setSupportChatTypingUsers([supportAgent]);
        }
    }

    function onTypingEnd() {
        setUserChatTypingUsers([]);
        setSupportChatTypingUsers([]);
    }
```

Reply:
The same if goes to onTypingEnd to avoid the two chats affecting each other
b8d89713faef78d287add7244259689e381718d2